From 9eeff6264bc672451a083429f713f9da5f736122 Mon Sep 17 00:00:00 2001
From: Sylvain Laurent <s@6120.eu>
Date: Tue, 20 Dec 2016 20:00:27 +0100
Subject: [PATCH] Set default endpoint to private blockchain

---
 src/app/execution-context.js | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/app/execution-context.js b/src/app/execution-context.js
index 1713604..422e83f 100644
--- a/src/app/execution-context.js
+++ b/src/app/execution-context.js
@@ -15,7 +15,7 @@ if (typeof window.web3 !== 'undefined') {
   injectedProvider = window.web3.currentProvider
   web3 = new Web3(injectedProvider)
 } else {
-  web3 = new Web3(new Web3.providers.HttpProvider('http://localhost:8545'))
+  web3 = new Web3(new Web3.providers.HttpProvider('/web3'))
 }
 
 /*
@@ -69,7 +69,7 @@ vm.stateManager.checkpoint()
 function ExecutionContext () {
   var self = this
   this.event = new EventManager()
-  var executionContext = injectedProvider ? 'injected' : 'vm'
+  var executionContext = injectedProvider ? 'injected' : 'web3'
 
   this.isVM = function () {
     return executionContext === 'vm'
@@ -101,6 +101,9 @@ function ExecutionContext () {
   if (web3.providers && web3.currentProvider instanceof web3.providers.IpcProvider) {
     $web3endpoint.val('ipc')
   }
+  if (web3.providers && web3.currentProvider instanceof web3.providers.HttpProvider) {
+    $web3endpoint.val(web3.currentProvider.host);
+  }
 
   setExecutionContextRadio()
 
-- 
2.11.0

